### ISSUE TITLE
InvalidAuthenticationToken Error

### ISSUE THREAD

**Issue by yeshwanthjagannath** [MicrosoftDocs/office-docs-powershell#156]
### I am trying to run a script from a .ps1 file. The script looks as follows:
$UserName="myusername@domain.com"
$password = "mypassword"
$password = ConvertTo-SecureString $password -force -asPlainText
$creds = New-Object System.Management.Automation.PSCredential ($UserName, $password)
$credobject = Connect-MicrosoftTeams -Credential $creds
$teamname = "TestPS5"
$group = New-Team -DisplayName $teamname -AccessType Private
$group.GroupId
-------------------------------------------------------------------------------------------------------------
But the script runs fine sometimes and gives me the below error majority of the times:

### New-Team : Error occurred while executing
Code: InvalidAuthenticationToken
Message: Bearer access token is empty.
InnerError:
  RequestId: 80705785-a503-4844-9a09-2254233d6315
  DateTimeStamp: 2018-02-12T16:28:40
HttpStatusCode: InvalidAuthenticationToken
At C:\Users\yjagannathx070430\Documents\Yeshwanth_Files\O365\31JAN2018\TeamsTest.PS1:8 char:10
+ $group = New-Team -DisplayName $teamname -AccessType Private
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-Team], ApiException
    + FullyQualifiedErrorId : Microsoft.TeamsCmdlets.PowerShell.Custom.ErrorHandling.ApiException,Microsoft.TeamsCmdle
   ts.PowerShell.Custom.NewTeam

Any pointers why this is so inconsistent? Why does create the teams sometimes and why does it fail most times?

**Comment by kenwith** [comment]
@yeshwanthjagannath - what version of the Teams cmdlets are you using? I asked the PM that owns Teams and she said that the 0.9.1 release has some retry logic built in. You can see the version with the command    get-module -listavailable   and then look for MicrosoftTeams in the Name column and the Version is just to the left.

**Comment by kenwith** [comment]
Also, not sure if this is the same issue but adding a link: https://github.com/MicrosoftDocs/office-docs-powershell/issues/147

**Comment by yeshwanthjagannath** [comment]
@kenwith I am using the 0.9.1 release.

**Comment by kenwith** [comment]
@SamCosby - could these be related to the 147 issue? (saml version) My gut tells me it isn't though because some succeed and some fail. Any thoughts @islubin ?

**Comment by kenwith** [comment]
@yeshwanthjagannath  - it might be worth also opening a support ticket in your tenant in parallel. That way a dedicated resource to track down. I am still working to figure out too and we can update Notes section in docs with resolution.

**Comment by yeshwanthjagannath** [comment]
@kenwith Yes, i have already done that and i have been in touch with the support focal.

**Comment by islubin** [comment]
@yeshwanthjagannath - are these calls failing within short periods of time of successes (ie one succeeds and then 30 seconds later, another fails?)  

On a separate note - you shouldn't be trying to combine providing a teamname with an existing group conversion.  If you're converting a group, it already has a display name.

**Comment by yeshwanthjagannath** [comment]
@islubin - there is no particular pattern of failure, but it succeeds the first few times and then fails and just continues to fail until i reboot and happen to try again. If i am lucky it fires off fine, else it continues to fail.
And, no i am not converting an existing group. I am passing the teamname as a parameter to the script to be the name of the Team we are trying to create through script.

**Comment by islubin** [comment]
Sorry - I read your PS snippet too fast and zeroed in on $group.GroupId, which I thought you were feeding into New-Team. :)  

Adding in @nkramer

**Comment by SamCosby** [comment]
@kenwith, doesn't look like the same failure, though, I'd like to validate when this does fail as what's happening via Fiddler. Could be something failing against the IDP of some sort.

**Comment by yeshwanthjagannath** [comment]
@SamCosby Thanks for reaching out over email and getting a working session on. As discussed, this continues to be an open issue at this time against the MS Graph API.

For the benefit of the rest on this thread:
We are looking at utilizing Azure runbooks (Azure automation services) to run the MicrsoftTeams powershell, which has been more promising at this point. However, we notice a lot of lag, approx. 1-2 hours before the team can appear on the Teams app.

Secondly, there are no notifications being sent out to the users, when added to a team through PowerShell - any inputs on this?

**Comment by yeshwanthjagannath** [comment]
Guys,
We tried an alternate approach on this by executing the PowerShell on Azure Runbooks and invoking the runbook through MS Flow, that works like a charm and we do not get errors of any sort. So for now, until the bearer access token issue can be fixed, we will proceeding with hosting our PowerShell as an Azure Runbook. Just thought i would share here for the benefit of the community.

**Comment by kenwith** [comment]
@SamCosby @yeshwanthjagannath  - this is really great info. I wonder if there is something we could add to the Notes section of the connect cmdlet? Should we leave this issue open for now?

**Comment by kenwith** [comment]
Closing this one out. Please re-open if needed.
